ui/src/components/TrafficPage.tsx

```diff
@@ -235,17 +242,41 @@ export const TrafficPage: React.FC<TrafficPageProps> = ({onTLSDetected, setAnaly
     return (
         <div className="TrafficPage">
             <div className="TrafficPageHeader">
-                <img className="playPauseIcon" style={{visibility: wsConnection === WsConnectionStatus.Connected ? "visible" : "hidden"}} alt="pause"
-                    src={pauseIcon} onClick={toggleConnection}/>
-                <img className="playPauseIcon" style={{position: "absolute", visibility: wsConnection === WsConnectionStatus.Connected ? "hidden" : "visible"}} alt="play"
-                    src={playIcon} onClick={toggleConnection}/>
-                <div className="connectionText">
-                    {getConnectionTitle()}
-                    <div className={"indicatorContainer " + getConnectionStatusClass(true)}>
-                        <div className={"indicator " + getConnectionStatusClass(false)}/>
-                    </div>
-                </div>
+              <div className="TrafficPageStreamStatus">
+                  <img className="playPauseIcon" style={{visibility: wsConnection === WsConnectionStatus.Connected ? "visible" : "hidden"}} alt="pause"
+                      src={pauseIcon} onClick={toggleConnection}/>
+                  <img className="playPauseIcon" style={{position: "absolute", visibility: wsConnection === WsConnectionStatus.Connected ? "hidden" : "visible"}} alt="play"
+                      src={playIcon} onClick={toggleConnection}/>
+                  <div className="connectionText">
+                      {getConnectionTitle()}
+                      <div className={"indicatorContainer " + getConnectionStatusClass(true)}>
+                          <div className={"indicator " + getConnectionStatusClass(false)}/>
+                      </div>
+                  </div>
+              </div>
+              {window["isOasEnabled"] && <div>
+                <Button
+                  type="submit"
+                  variant="contained"
+                  style={{
+                    margin: "2px 0px 0px 0px",
+                    backgroundColor: variables.blueColor,
+                    fontWeight: 600,
+                    borderRadius: "4px",
+                    color: "#fff",
+                    textTransform: "none",
+                  }}
+                  onClick={handleOpenModal}
+                >
+                  Show OAS
+                </Button>
+              </div>}
             </div>
+            {window["isOasEnabled"] && <OasModal
+              openModal={openOasModal}
+              handleCloseModal={handleCloseModal}
+              entries={entries}
```

Comment:
you don't need to pass entries - you can take it from recoil at OasModal component